### PR TITLE
Default dbt asset checks on

### PR DESCRIPTION
**Pull request by johannkm**
Docs changes coming

### CODE REVIEW

**Comment by johannkm** [comment]
* **#20812** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20812?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @johannkm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rexledesma — COMMENTED** [review]
I will wait for the docs to be written before approving this PR.

Some things that we should mention in the docs for Asset Checks GA:
- Singular tests are now supported. We'll need to document the new syntax for declaring the primary asset key on a singular test with multiple references: https://github.com/dagster-io/dagster/pull/20801. 
- Singular tests with multiple dependencies that are not configured with a primary asset key will emit `AssetObservation`.
- Configuring a primary asset key on a singular test requires `dbt-core>=1.6`.

**Review by rexledesma — DISMISSED** [review]
Requesting changes to put this back in your queue

**Comment by johannkm** [comment]
docs pr https://github.com/dagster-io/dagster/pull/20914

File: python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py [hunk 1]
```diff
@@ -1321,7 +1321,7 @@ def get_subset_selection_for_context(
 
     # note that this will always be true if checks are disabled
     if selected_checks_target_selected_assets and all_checks_included_for_selected_assets:
-        selected_dbt_resources = selected_dbt_non_test_resources + selected_dbt_tests
```
Comment (johannkm) [inline python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py:1324]: glad I caught this- bug from https://github.com/dagster-io/dagster/pull/ that had us still listing checks explicitly. Tested by https://github.com/dagster-io/dagster/blob/johann/03-28-Default_dbt_asset_checks_on/python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_resources_v2.py#L375 now that checks are on by default

Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py:1324]: The pull request you linked in broken -- could you link the correct one?

Comment (johannkm) [reply python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py:1324]: oops, https://github.com/dagster-io/dagster/pull/20568

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_decorator.py [hunk 2]
```diff
@@ -244,7 +244,10 @@ def my_dbt_assets(): ...
     expected_io_manager_key = DEFAULT_IO_MANAGER_KEY if io_manager_key is None else io_manager_key
 
     for output_def in my_dbt_assets.node_def.output_defs:
-        assert output_def.io_manager_key == expected_io_manager_key
+        # asset checks don't use io managers
+        if output_def.name not in my_dbt_assets.check_specs_by_output_name:
+            assert output_def.name in my_dbt_assets.keys_by_output_name
+            assert output_def.io_manager_key == expected_io_manager_key
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_decorator.py:250]: let's invert the logic to make it explicit that we're ignoring asset checks here
```suggestion
        # asset checks don't use io managers
        if output_def.name in my_dbt_assets.check_specs_by_output_name:
            continue

        assert output_def.io_manager_key == expected_io_manager_key
```

File: python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py (hunk 1 above)
Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py:1324]: Could you add an explicit test in `test_asset_checks.py`? One that would have failed master without this change.

We're relying on the implicit assumption that asset checks is enabled (which is why we caught this in `test_resources_v2.py` in this PR. 

We should make our implicit assumption explicit in the test case. [1]

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py [hunk 3]
```diff
@@ -26,8 +26,9 @@
 
 pytest.importorskip("dbt.version", "1.6")
 
-dagster_dbt_translator_with_checks = DagsterDbtTranslator(
-    settings=DagsterDbtTranslatorSettings(enable_asset_checks=True)
+
+dagster_dbt_translator_without_checks = DagsterDbtTranslator(
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py:30]: [1]

I think we should keep this object for tests.

The tests that I would add here are:
- We test that a default `DagsterDbtTranslator` without settings passed in results in asset checks being enabled
- Introduce a separate `dagster_dbt_translator_without_checks` and test that asset checks are not enabled when that object is passed. 
- We keep the existing tests as they are. This way, we're not relying on the implicit behavior to do our testing for us.

**Review by rexledesma — DISMISSED** [review]

File: python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py (hunk 1 above)
Comment (johannkm) [reply python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py:1324]: moved the bugfix to a separate pr with explicit tests: https://github.com/dagster-io/dagster/pull/20944

**Review by rexledesma — APPROVED** [review]

**Pull request merged**